modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/internal/security/permission/resource/MBThreadModelResourcePermission.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.message.boards.internal.security.permission.resource;
+
+import com.liferay.message.boards.model.MBMessage;
+import com.liferay.message.boards.model.MBThread;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.security.auth.PrincipalException;
+import com.liferay.portal.kernel.security.permission.PermissionChecker;
+import com.liferay.portal.kernel.security.permission.resource.ModelResourcePermission;
+import com.liferay.portal.kernel.security.permission.resource.PortletResourcePermission;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Jiaxu Wei
+ */
+@Component(
+	property = "model.class.name=com.liferay.message.boards.model.MBThread",
+	service = ModelResourcePermission.class
+)
+public class MBThreadModelResourcePermission
+	implements ModelResourcePermission<MBThread> {
+
+	@Override
+	public void check(
+			PermissionChecker permissionChecker, long messageId,
```

Comment:
@W2jx The `ModelResourcePermission<T>` interface names the `long` parameter `primaryKey`, we should assume that the `primaryKey` is the pk of the `T`.

Do you see this being a common pattern in portal? If not, please change it to `threadId` and use it as `threadId`; you can always get the root message id from the thread id though.

Reply:
In this commit message, you say `sup-class`. It's confusing. I thought you meant `sub-class` and made a typo. Please always say the full word, "super class".